towhee/hub/file_manager.py

```diff
@@ -228,12 +227,11 @@ def _cache_name(self, name: str, author: str, branch: str):
         try:
             file_name, file_type = name.split('.')
         except ValueError:
-            return name.replace('-', '_') + '&' + author + '$' + branch
-
+            return author + '/' + name.replace('-', '_') + '/' + branch
         # If pointing directly to pipeline or operator file.
         if file_type in ['py', 'yaml']:
-            return file_name.replace('-', '_') + '&' + author + '$' + branch
-        # If not pointing to a pipeline or operator file.
+            return author + '/' + file_name.replace('-', '_') + '/' + branch
```

Comment:
We should use `pathlib` to join path, for `towhee` need to support windows, Linux and Mac.

Reply:
fixed